backend/src/api/game/challenge.rs

```diff
@@ -19,34 +68,107 @@ pub struct NewChallengeRequest {
     // cannot be played first.
     pub tournament_queen_rule: bool,
 
+    // The challenger's color choice
+    pub color_choice: ColorChoice,
+
     pub game_type: GameType,
 }
 
 #[derive(Serialize)]
-pub struct NewChallengeResponse {
-    challenge_url: String,
+#[serde(rename_all = "camelCase")]
+pub struct GameChallengeResponse {
+    pub id: Uuid,
+    pub challenger: User,
+    pub game_type: GameType,
+    pub ranked: bool,
+    pub public: bool,
+    pub tournament_queen_rule: bool,
+    pub color_choice: ColorChoice,
+    pub created_at: DateTime<Utc>,
+}
+
+impl GameChallengeResponse {
+    pub async fn from_model(challenge: &GameChallenge, pool: &DbPool) -> Result<Self, ServerError> {
+        let challenger = match challenge.get_challenger(&pool).await {
+            Ok(challenger) => challenger,
+            Err(diesel::result::Error::NotFound) => {
+                let uid = challenge.challenger_uid.clone();
+                return Err(ChallengeError::MissingChallenger(uid).into());
+            }
+            Err(err) => return Err(err.into()),
+        };
+        GameChallengeResponse::from_model_with_user(challenge, challenger)
+    }
+
+    pub fn from_model_with_user(
+        challenge: &GameChallenge,
+        challenger: User,
+    ) -> Result<Self, ServerError> {
+        let game_type: GameType = challenge
+            .game_type
```

Comment:
If they implement a `from`like our other errors you can just use `?` on them

Reply:
`GameError` already has a `From` implementation for our ServerError type, but it becomes a 400 Bad Request, which wouldn't be right for an error within our database -- hence why i manually specify that this is an internal server error